include/index_factory.h

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Is the idea to have an IndexFactory and a DynamicIndexFactory, or should we figure out how the dynamic indices come into here?  Is it two enums, or a singular set of enums like `DYNAMIC_DISK` and `STATIC_MEMORY`.  Also, how does Disk and SSD differ?

Reply:
I was thinking more like, we have a flag if the build is dynamic or not, while creating the Instance of Disk or Memory Index we can use this flag to appropriately instantiate the class.  Let me look into this, thanks for pointing this out.